homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from asyncio.exceptions import CancelledError
+from typing import Any
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_can_discover_devices() -> bool:
+    """Return if there are devices that can be discovered."""
+    future_event: Future[None] = asyncio.get_running_loop().create_future()
+
+    @callback
+    def _async_found(_):
+        """Handle a discovered device - only need to do this once so."""
+
+        if not future_event.done():
+            future_event.set_result(None)
+
+    async with WeatherFlowListener() as client, asyncio.timeout(10):
+        try:
+            client.on(EVENT_DEVICE_DISCOVERED, _async_found)
+            await future_event
+        except asyncio.TimeoutError:
+            return False
+        except CancelledError:
+            raise ListenerError
+
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        # Only allow a single instance of integration since the listener
+        # will pick up all devices on the network and we don't want to
+        # create multiple entries.
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        errors = {}
+        try:
+            await _async_can_discover_devices()
```

Comment:
It looks like the return result is no longer being checked?

Reply:
Correct ... when I was on discord with @joostlek  - we figured because a timeout/canceleld error gets raised (still dont know why cancelled) we dont actually need to check the result .. just need to exit cleanly (found devices) or exit poorly (didn't).

(is that ok?)

Reply:
Please remove the return value since nothing care about it, and its confusing to return something you are always going to throw away in all instances

Reply:
I mean, if we don't use the result (and we know the way it is checked is good enough)

Reply:
This doesn't seem right to remove the check for the return value since if it times out, nothing checks for False and its going to create an entry anyways even if it didn't see a packet.

You could instead change the flow of `_async_can_discover_devices` to not trap `asyncio.TimeoutError` and trap it here instead and abort with no devices found.

Reply:
https://github.com/home-assistant/core/pull/75530/files#diff-b4784835252d4458a75e882760fd64e2648074ad5d671b620dc84a02eeb30b16R18

`no_device_found` is in strings but there is no way for it to open.

We need to fix that and add a test for it